src/mixins/filing-template-mixin.ts

```diff
@@ -364,16 +359,24 @@ export default class FilingTemplateMixin extends Mixins(DateMixin) {
 
     if (this.hasPeopleAndRolesChanged) {
       let parties = this.getPeopleAndRoles
-      // if filing and paying, filter out removed entities and omit the 'action' properties
+      // if filing and paying, filter out removed entities and omit the 'actions' properties
       if (!isDraft) {
         // Filter out parties actions
-        parties = parties.filter(x => !x.action.includes(ActionTypes.REMOVED))
+        parties = parties.filter(x => !x.actions?.includes(ActionTypes.REMOVED))
           .map((x) => {
-            const { action, ...rest } = x
+            // Remove null or undefined properties to adhere to schema.
+            // ie taxId is an optional property but schema doesn't accept defined empty properties
```

Comment:
I shudder when I see `Object.keys` -- it should not be needed.
You might be able to just set known keys (that the API doesn't like) to undefined. Remember that trick about JS object <-> JSON where it doesn't include undefined properties?

Reply:
Hmm i'm not sure of the trick. 
In this case it is the `keys` that we want to omit, so if there is another way i'm all for it. 
I do agree, this issue is a result of the API returned a defined key with an empty value, which in turn the Schema does NOT like when returned back to the api. 

Was a pain to find 😵

Reply:
The trick is, if you set a prop to undefined then it doesn't get included in the JSON.

Earlier today I did this:
```
taxId: this.getRegistration.businessNumber || undefined // can't be null
```

Your code searches for any empty key. But if you knew which keys need to be stripped you could just assign undefined to them.

Reply:
This UI doesn't assign these values, purely receive from API, displays them and returns them. 
This could change once we use business look however. 

Would the above statement resolve to `taxId:  ''` or remove the property entirely from the object?

Reply:
We also can't eliminate it entirely because the api sometimes provides it.

Reply:
Honestly, this could be a schema issue, if it's going to provide empty properties, it should probably pass validation to receive it as such, i'm not sure.

Reply:
So you're saying, you can include values provided by the API but not the others?

Yes, since `''` is falsy, `|| undefined` will assign undefined to that variable. That will remove it entirely from the JSON that JS magically generates from this object.

Reply:
Obviously your code works. I'm trying to understand why it's needed and whether it can be less Object-specific, but maybe it's not worth further discussion today. Just resolve this conversation if you agree :)

Reply:
> We also can't eliminate it entirely because the api sometimes provides it.

That's where the `||` comes in (unfortunately that will also catch legitimate false values and numerical 0). You could check for "null or ``' and then assign undefined (instead of deleting).

At this point I'm not sure which is going to be cleaner, so again, I propose we just leave this alone.

For now. << kidding, haha

Reply:
haha totally, let's discuss on a call next week!
Thank you :)